### ISSUE TITLE
2 formatters used together don't make it to userspace

### ISSUE THREAD

**Issue by pimentel** [talonvoice/talon#44]
uttering `snake the test smash no` does not make it to the FormatText() function in `std.py`

Here is the debug.py:

```
2018-05-27 23:44:45    IO phrase {'cmd': 'p.begin', 'grammar': 'sleepy', 'ts': 78855563682}
2018-05-27 23:44:45    IO phrase {'cmd': 'p.begin', 'grammar': 'talon', 'ts': 78855563734}
2018-05-27 23:44:47    IO phrase {'cmd': 'p.hypothesis', 'grammar': 'sleepy', 'phrase': ['snake'],
'words': [{'word': 'snake', 'id': 0, 'rule': 0, 'start': 1527489885566, 'end': 1527489886226}], 'ts': 78857130829}
2018-05-27 23:44:47    IO phrase {'cmd': 'p.hypothesis', 'grammar': 'talon', 'phrase': ['snake'], 'words': [{'word': 'snake', 'id': 0, 'rule': 0, 'start': 1527489885566, 'end': 1527489886226}], 'ts': 78857130887}
2018-05-27 23:44:47    IO phrase {'cmd': 'p.hypothesis', 'grammar': 'sleepy', 'phrase': ['snake', 'the'], 'words': [{'word': 'snake', 'id': 0, 'rule': 0, 'start': 1527489885566, 'end': 1527489886226}, {'word': 'the', 'id': 0, 'rule': 0, 'start': 1527489886226, 'end': 1527489886346}], 'ts': 78857345543}
2018-05-27 23:44:47    IO phrase {'cmd': 'p.hypothesis', 'grammar': 'talon', 'phrase': ['snake', 'the'], 'words': [{'word': 'snake', 'id': 0, 'rule': 0, 'start': 1527489885566, 'end': 1527489886226}, {'word': 'the', 'id': 0, 'rule': 0, 'start': 1527489886226, 'end': 1527489886346}], 'ts': 78857345615}
2018-05-27 23:44:47    IO phrase {'cmd': 'p.end', 'grammar': 'talon', 'phrase': ['snake', 'the', 'test', 'smash', 'no'], 'words': [{'word': 'snake', 'id': 210, 'rule': 600, 'start': 1527489885546, 'end': 1527489886226}, {'word': 'the', 'id': 377, 'rule': 1000000, 'start': 1527489886226, 'end': 1527489886346}, {'word': 'test', 'id': 0, 'rule': 1000000, 'start': 1527489886346, 'end': 1527489886706}, {'word': 'smash', 'id': 211, 'rule': 600, 'start': 1527489886706, 'end': 1527489887126}, {'word': 'no', 'id': 83, 'rule': 1000000, 'start': 1527489887126, 'end': 1527489887716}], 'ts': 78857402240, 'path': [{'rule': 'input.keymap__dunder___camel___snake___smash___spine___title___dubstring___string___padded___rot_thirteen___dotsway___yellsnik___criff___yeller___thrack___quattro____dgndictation_', 'words': [{'word': 'snake', 'start': 1527489885546, 'end': 1527489886226}, {'rule': 'dgndictation', 'words': [{'word': 'the', 'start': 1527489886226, 'end': 1527489886346}, {'word': 'test',
'start': 1527489886346, 'end': 1527489886706}]}, {'word': 'the', 'start': 1527489886226, 'end': 1527489886346}]}, {'rule': 'dgndictation', 'words': [{'word': 'no', 'start': 1527489887126, 'end': 1527489887716}]}]}
2018-05-27 23:44:47    IO bad rule dgndictation
2018-05-27 23:44:47 ERROR cb error <class 'talon.engine.Engine'> topic="phrase" cb=<bound method ContextGroup.on_phrase of <talon.voice.ContextGroup object at 0x0000000110480800>>
Traceback (most recent call last):
  File "/Applications/Talon.app/Contents/Resources/pypy/site-packages/talon/dispatch.py", line 58,
in safe_call
    return self.call(cb, args)
  File "/Applications/Talon.app/Contents/Resources/pypy/site-packages/talon/dispatch.py", line 54,
in call
    return cb(*args)
  File "/Applications/Talon.app/Contents/Resources/pypy/site-packages/talon/voice.py", line 194, in on_phrase
    prefix, rule._name = rule._name.split('.', 1)
ValueError: not enough values to unpack (expected 2, got 1)
2018-05-27 23:44:47    IO phrase {'cmd': 'p.end', 'grammar': 'sleepy', 'phrase': [], 'ts': 78857402281, 'path': []}
```

**Comment by pimentel** [comment]
possible duplicate of: https://github.com/talonvoice/talon/issues/42

**Comment by lunixbochs** [comment]
Fixed in 0.0.8 with parser rewrite, now outputs:

the_testno
